internals/src/const_tools.rs

```diff
@@ -0,0 +1,92 @@
+//! Contains tools (workarounds) to make implementing `const fn`s easier.
+
+/// Copies first `$len` bytes from `$slice` and returns them as an array.
+///
+/// Returns `None` if `$len > $slice.len()`. `$len` must be (obviously) statically known.
+/// Calling from non-const context doesn't affect performance.
+#[macro_export]
+macro_rules! copy_byte_array_from_slice {
+    ($slice:expr, $len:expr) => {
+        if $len != $slice.len() {
+            None
```

Comment:
Well, it does match the name - array. :) But IIRC the reason is I wrote this a long time ago and then I decided to change it but changed it wrong. The whole `const_tools` module is just taken from my previous PR because I wanted the methods to be `const`.